### DISCUSSION TITLE
How to enable multifocusing with text field and Popover list

### DISCUSSION THREAD

**Discussion by candidia** [ariakit/ariakit#2566] [Questions] [answered]
I'm composing Popover with Lexical editor and the only thing I haven't figure out is how I can replicate the behavior where you are typing and can arrow key through the popover while still typing. If I use the default autoFocusOnShow on the Popover, it takes away focus from the input (which is worth mentioning is not a real input; Lexical uses a div with the contentEditable prop). I used Popover because I need to use anchorRef to position the popover, which doesn't seem possible with Select or Combobox.

I'm aware of this example https://ariakit.org/examples/combobox-textarea but it doesn't seem very composable with Lexical's plugin pattern. If you care to dive more into this, I'm specifically using Lexical's LexicalTypeaheadMenuPlugin, which has no documentation but has examples like https://github.com/facebook/lexical/blob/main/packages/lexical-playground/src/plugins/MentionsPlugin/index.tsx and https://github.com/facebook/lexical/blob/main/packages/lexical-playground/src/plugins/ComponentPickerPlugin/index.tsx I want to use ariakit instead of the native positioning in there though because it has issues with its positioning https://github.com/facebook/lexical/issues/3834

I'm not sure whether to make an example with Lexical, or if this can be boiled down to a more generic issue of how this would behave with a Popover and any contentEditable element, so I'll hold off. I guess I'm just wondering how Combobox does it, ie not steal focus from the input while allowing you to arrow key through its items?

**Comment by diegohaz** [comment]
There's an old example using Combobox and Lexical here: https://ariakit-kmqava9a4-ariakit.vercel.app/examples/combobox-lexical

It's using older versions of Ariakit and Lexical, but can give you some insights.

**Reply by candidia** [reply]
Thank you! I notice it uses the same pattern of having the input be the combobox, and ariakit therefore takes over the responsibility of listening for the trigger, but I dislike that because lexical should really be in charge and I don't always want to render the combobox, for example I may want to lazy load it. I now have a SelectPopover being rendered when lexical detects a trigger, but my only issue is the focussing. The Popover steals the focus from the input. I've been exploring props like `composite` `autoFocus` `includesBaseElement` `activeId` `virtualFocus`, which all seem related but I can't quite get it. I might have to look into the Combobox internals to see how it maintains focus on the input while making the popover arrow key navigable

**Reply by diegohaz** [reply, marked as answer]
Combobox uses the Composite component with `virtualFocus` set to `true`: https://codesandbox.io/s/ariakit-composite-with-virtualfocus-k4jfj5

**Reply by candidia** [reply]
@diegohaz coming back with a repro of my issue this time in tiptap which could simplify things and wondering if you know if there's a way to get it working from here https://stackblitz.com/edit/stackblitz-starters-tzkrwx?file=src%2FApp.tsx,src%2Fstyle.css

**Reply by diegohaz** [reply]
If you want to take advantage of Ariakit, you'll have to render your input/textarea/contenteditable element with a composite component like `Combobox`. Ariakit doesn't have to control the trigger, and you can lazy load the popover part, but Ariakit needs to pass props to the input element.

**Reply by candidia** [reply]
Awesome thanks for the lead. Will give it a go.

**Reply by candidia** [reply]
https://stackblitz.com/edit/stackblitz-starters-j5p2li got the tiptap contentEditable being rendered as a Composite but focus isn't being shared. It seems like the composite is being rendered as a parent instead of the element itself though.

**Reply by diegohaz** [reply]
You're appending the contenteditable node as a child of the composite element: https://stackblitz.com/edit/stackblitz-starters-j5p2li?file=src%2FEditorContent.tsx%3AL71

**Reply by candidia** [reply]
I'm not sure how to merge the two elements

**Reply by candidia** [reply]
For now I'm just using floating-ui directly. There are a couple drawbacks like not being able to use the ariakit CSS variables for width and height, but I don't think there is currently a way to make ariakit work with tiptap, both are trying to control the focus.
